File: backend/data-transformer/src/models/secondary/item.ts
import { Expose, Transform } from "class-transformer";

export class Item {
    @Expose() refSlug!: string;
    img!: string;
    @Transform((obj) => {
        if (typeof obj === "number") {
            const newValue = Number.parseInt(obj);
            Object.assign(obj, newValue);
            return newValue;
        }
    })
    @Expose()
    level!: number;
    name!: string;
    uuid!: string;
}

File: backend/src/models/item.rs
use serde::{Deserialize, Serialize};
use specta::Type;

#[derive(Serialize, Deserialize, Debug, Type)]
#[serde(rename_all = "camelCase", deny_unknown_fields)]
pub struct Item {
    ref_slug: Option<String>,
    img: Option<String>,
    level: Option<u8>,
    name: String,
    uuid: String,
}
